js/app/utils.js

```diff
@@ -36,19 +36,32 @@ let _ = Gettext.dgettext.bind(null, Config.GETTEXT_PACKAGE);
 /* Not part of public API. Changes @widget's GdkWindow to have the 'hand' cursor
 indicating a clickable UI element. */
 function set_hand_cursor_on_widget(widget) {
+    if (widget._hand_cursor_handlers)
+        return;
+
     widget.add_events(Gdk.EventMask.ENTER_NOTIFY_MASK |
         Gdk.EventMask.LEAVE_NOTIFY_MASK);
 
-    widget.connect('enter-notify-event', function (widget) {
+    let enter = widget.connect('enter-notify-event', function (widget) {
         let cursor = Gdk.Cursor.new_for_display(Gdk.Display.get_default(),
             Gdk.CursorType.HAND1);
         widget.window.set_cursor(cursor);
         return Gdk.EVENT_PROPAGATE;
     });
-    widget.connect('leave-notify-event', function (widget) {
+    let leave = widget.connect('leave-notify-event', function (widget) {
         widget.window.set_cursor(null);
         return Gdk.EVENT_PROPAGATE;
     });
+
+    widget._hand_cursor_handlers = { enter_id: enter, leave_id: leave };
```

Comment:
drive-by comment because I'm always on the lookout for opportunities to use the cool new ES6 syntax :wink: 
You can now do stuff like this:
```js
let enter_id = ...;
let leave_id = ...;
widget._hand_cursor_handlers = { enter_id, leave_id };
```